.github/workflows/pants.yaml

```diff
@@ -0,0 +1,62 @@
+---
+name: Validate Pants Metadata
+
+on:
+  # temporarily only allow manual runs until we have BUILD files and lockfiles
+  workflow_dispatch:
+  #push:
+  #  branches:
+  #    # only on merges to master branch
+  #    - master
+  #    # and version branches, which only include minor versions (eg: v3.4)
+  #    - v[0-9]+.[0-9]+
+  #  tags:
+  #    # also version tags, which include bugfix releases (eg: v3.4.0)
+  #    - v[0-9]+.[0-9]+.[0-9]+
+  #pull_request:
+  #  type: [opened, reopened, edited]
+  #  branches:
+  #    # Only for PRs targeting those branches
+  #    - master
+  #    - v[0-9]+.[0-9]+
+
+jobs:
+  pants-tailor:
+    name: Make sure pants BUILD files are up-to-date
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          # a test uses a submodule, and pants needs access to it to calculate deps.
+          submodules: 'true'
+
+      - name: Initialize Pants and its GHA caches
+        uses: pantsbuild/actions/init-pants@c0ce05ee4ba288bb2a729a2b77294e9cb6ab66f7
+        # This action adds an env var to make pants use both pants.ci.toml & pants.toml.
+        # This action also creates 3 GHA caches (1 is optional).
+        # - `pants-setup` has the bootsrapped pants install
+        # - `pants-named-caches` has pip/wheel and PEX caches
+        # - `pants-lmdb-store` has the fine-grained process cache.
+        #   If we ever use a remote cache, then we can drop this.
+        #   Otherwise, we may need an additional workflow or job to delete old caches
+        #   if they are not expiring fast enough, and we hit the GHA 10GB per repo max.
+        with:
+          # To ignore a bad cache, bump the cache* integer.
+          gha-cache-key: cache0-BUILD
+          # This hash should include all of our lockfiles so that the pip/pex caches
+          # get invalidated on any transitive dependency update.
+          named-caches-hash: ${{ hashFiles('requirements.txt' }}
+          # enable the optional lmdb_store cache since we're not using remote caching.
+          cache-lmdb-store: 'true'
```

Comment:
If we experience any issues with the 10GB GHA cache limit (a per-repo limit), we may need to add a workflow that periodically clears out older unused caches - Github expires unused caches after 7 days, but we might want a shorter timespan, as this cache is a per-commit cache. We can also bump the `gha-cache-key` input var if the cache gets too big to ignore it and start fresh.